src/Our.Umbraco.InnerContent/Web/Controllers/InnerContentApiController.cs

```diff
@@ -32,18 +32,19 @@ public IEnumerable<object> GetAllContentTypes()
         [HttpGet]
         public IEnumerable<object> GetContentTypesByGuid([ModelBinder] Guid[] guids)
         {
-            return Services.ContentTypeService.GetAllContentTypes()
-                .Where(x => guids == null || guids.Contains(x.Key))
-                .OrderBy(x => x.SortOrder)
-                .Select(x => new
-                {
-                    id = x.Id,
-                    guid = x.Key,
-                    name = x.Name,
-                    alias = x.Alias,
-                    icon = string.IsNullOrWhiteSpace(x.Icon) || x.Icon == ".sprTreeFolder" ? "icon-folder" : x.Icon,
-                    tabs = x.CompositionPropertyGroups.Select(y => y.Name).Distinct()
-                });
+            var contentTypes = Services.ContentTypeService.GetAllContentTypes(guids).OrderBy(x => x.SortOrder).ToList();
+            var blueprints = Services.ContentService.GetBlueprintsForContentTypes(contentTypes.Select(x => x.Id).ToArray()).ToArray();
+
+            // NOTE: Using an anonymous class, as the `ContentTypeBasic` type is heavier than what we need (for our requirements)
+            return contentTypes.Select(ct => new
+            {
+                name = ct.Name, // TODO: localize the name (in case of dictionary items), e.g. `localizedTextService.UmbracoDictionaryTranslate`
+                description = ct.Description, // TODO: localize the description (in case of dictionary items), e.g. `localizedTextService.UmbracoDictionaryTranslate`
+                key = ct.Key,
```

Comment:
Change from "guid" to "key", need to check if this is every exposed outside of inner content as if so, this is a breaking change.

Reply:
We can revert back to `guid` if you prefer? But yeah a breaking-change.
It's not impacted Stacked Content - which has been our only known implementation.

Reply:
Hmm, we could really do with knowing if anyone else is using inner content yet. My gut says no, in which case, it probably doesn't matter if we change it.

Reply:
Well, there's nothing "official" (e.g. published on NuGet) 😃 

- http://nugetmusthaves.com/Dependency/Our.Umbraco.InnerContent
- http://nugetmusthaves.com/Dependency/Our.Umbraco.InnerContent.Core

Still, I'd like to "do the right thing" in terms of handling API changes.

Reply:
If that's the case, then maybe have a guid and a key?